src/bootstrap/format.rs

```diff
@@ -118,8 +118,19 @@ pub fn format(build: &Build, check: bool) {
         .to_path_buf();
     let src = build.src.clone();
     let (tx, rx): (SyncSender<PathBuf>, _) = std::sync::mpsc::sync_channel(128);
-    let walker =
-        WalkBuilder::new(src.clone()).types(matcher).overrides(ignore_fmt).build_parallel();
+    let walker = match paths.get(0) {
+        Some(first) => {
+            let mut walker = WalkBuilder::new(first);
+            for path in &paths[1..] {
+                walker.add(path);
```

Comment:
Hm. This doesn't do any verification of what the passed path is, which might be a problem. I'm not sure the ignore_fmt rules will work when not rooted at the build.src.clone() start; have you checked that e.g. passing src/test/ui doesn't change anything? If not, this isn't quite the right fix.

Reply:
> have you checked that e.g. passing src/test/ui doesn't change anything

I have now! :laughing: and there are no changes; I have a hard time believing *everything* there is formatted how rustfmt likes it, so I think we're good :)